### PR TITLE
feat: Add withdrawTo functionality to support exits to other addresses

### PR DESCRIPTION
**Pull request by QEDK**

### CODE REVIEW

File: contracts/examples/erc1155-transfer/FxERC1155ChildTunnel.sol [hunk 1]
```diff
@@ -7,124 +7,140 @@ import { Create2 } from "../..//lib/Create2.sol";
...
+	// root to child token
+	mapping(address => address) public rootToChildToken;
+	address public tokenTemplate;
+
+	constructor(address _fxChild, address _tokenTemplate) FxBaseChildTunnel(_fxChild) {
+		tokenTemplate = _tokenTemplate;
+		require(_isContract(_tokenTemplate), "Token template is not contract");
+	}
+
+	function withdraw(address childToken, uint256 id, uint256 amount, bytes memory data) public {
```
Comment (QEDK) [inline contracts/examples/erc1155-transfer/FxERC1155ChildTunnel.sol]: The changes are these 4 functions, and the internal `_withdraw` functions below. Had to reset the formatting here to be similar to all other contracts.

File: contracts/examples/erc721-transfer/FxERC721ChildTunnel.sol [hunk 2]
```diff
@@ -97,16 +102,42 @@ contract FxERC721ChildTunnel is FxBaseChildTunnel, Create2, IERC721Receiver {
...
+        address rootToken = childTokenContract.connectedToken();
+
+        // validate root and child token mapping
+        require(
+            childToken != address(0x0) && rootToken != address(0x0) && childToken == rootToChildToken[rootToken],
+            "FxERC721ChildTunnel: NO_MAPPED_TOKEN"
+        );
+
+        // withdraw tokens
+        childTokenContract.burn(tokenId);
```
Comment (imyourm8) [inline contracts/examples/erc721-transfer/FxERC721ChildTunnel.sol:135]: childTokenContract uses msg.sender internally? Other tokens use user's address to burn them, why does this contract uses the tunnel itself?

Comment (QEDK) [reply contracts/examples/erc721-transfer/FxERC721ChildTunnel.sol:135]: We are using OpenZeppelin implementations of the ERC token contracts, in the case of `ERC721` there is no such parameter, hence we don't have it either.

**Comment by jdkanani** [comment]
@QEDK can you please resolve the conflict?

**Comment by QEDK** [comment]
@jdkanani Done! 👌

**Comment by QEDK** [comment]
Merging after a conversation with @imyourm8 💯

**Pull request merged**